File: SDKs/sportsdata.io/nfl-v3-stats/1.0/python/src/sdk/models/operations/legacyboxscoresdeltacurrentweek.py
from dataclasses import dataclass, field
from typing import Any,List,Optional
from enum import Enum

class LegacyBoxScoresDeltaCurrentWeekFormatEnum(str, Enum):
    XML = "XML"
    JSON = "JSON"


@dataclass
class LegacyBoxScoresDeltaCurrentWeekPathParams:
    format: LegacyBoxScoresDeltaCurrentWeekFormatEnum = field(metadata={'path_param': { 'field_name': 'format', 'style': 'simple', 'explode': False }})
    minutes: str = field(metadata={'path_param': { 'field_name': 'minutes', 'style': 'simple', 'explode': False }})
    

@dataclass
class LegacyBoxScoresDeltaCurrentWeekRequest:
    path_params: LegacyBoxScoresDeltaCurrentWeekPathParams = field()
    

@dataclass
class LegacyBoxScoresDeltaCurrentWeekResponse:
    content_type: str = field()
    status_code: int = field()
    box_scores: Optional[List[Any]] = field(default=None)
    

File: SDKs/sportsdata.io/nfl-v3-stats/1.0/typescript/src/sdk/models/operations/legacyboxscoresdeltacurrentweek.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum LegacyBoxScoresDeltaCurrentWeekFormatEnum {
    Xml = "XML",
    Json = "JSON"
}


export class LegacyBoxScoresDeltaCurrentWeekPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=format" })
  format: LegacyBoxScoresDeltaCurrentWeekFormatEnum;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=minutes" })
  minutes: string;
}


export class LegacyBoxScoresDeltaCurrentWeekRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: LegacyBoxScoresDeltaCurrentWeekPathParams;
}


export class LegacyBoxScoresDeltaCurrentWeekResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  boxScores?: any[];

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
